### PR TITLE
Port overhang extra perimeters from PrusaSlicer

### PR DESCRIPTION
**Pull request by Noisyfox**
4512415a7e1ba542e73f154aaa16522b

#471

### CODE REVIEW

**Comment by Noisyfox** [comment]
2c1268db0d5b43bf6926c85dcbc7b2fb_720
62fcb221887671cbe6c0a80c9ef3f6d8_720

**Comment by tome9111991** [comment]
looks good
no extra perimeter:
no extra

with extra perimeter:
extra


Was evtl zum problem werden könnte für "extra perimeter on overhangs" wird der Bridge speed benutzt
in mein fall 50 mm/s

File: src/libslic3r/PerimeterGenerator.cpp [hunk 1]
```diff
@@ -854,6 +857,445 @@ void PerimeterGenerator::split_top_surfaces(const ExPolygons &orig_polygons, ExP
     //}
 }
 
+// find out if paths touch - at least one point of one path is within limit distance of second path
```
Comment (SoftFever) [inline src/libslic3r/PerimeterGenerator.cpp:860]: Could we add credit to the original author?
Something like this:
// Port "extra perimeters on overhangs" from PrusaSlicer. Original author: PavelMikus pavel.mikus.mail@seznam.cz

**Review by SoftFever — DISMISSED** [review]
LGTM.
Good to merge once the credits are added

**Comment by SoftFever** [comment]
> looks good no extra perimeter: no extra
> 
> with extra perimeter: extra
> 
> What could be a problem for "extra perimeter on overhangs" the bridge speed is used. Because it is a 100% overhang. in my case 50 mm/s

Feels a almost like a arc overhangs, lol

File: src/libslic3r/PerimeterGenerator.cpp (hunk 1 above)
Comment (Noisyfox) [reply src/libslic3r/PerimeterGenerator.cpp:861]: Done

**Review by SoftFever — APPROVED** [review]

**Comment by SoftFever** [comment]
> > Feels almost like arc overhangs, lol
> 
> It does, bacause they are based on the same principle.

True.
It's slightly different since the arc overhang is technically considered an infill.

**Comment by Noisyfox** [comment]
> What could be a problem for "extra perimeter on overhangs" the bridge speed is used.

BTW that seems to be a separate problem that not really related to this feature. Will look into that soon.

**Comment by tome9111991** [comment]
> > What could be a problem for "extra perimeter on overhangs" the bridge speed is used.
> 
> BTW that seems to be a separate problem that not really related to this feature. Will look into that soon.

yes is no issue from this feature.
the problem is that generally 100% overhangs use the bridge speed, it is also in the tooltip.
speed


Here a 100%  overhang it use the Bridge speed
bri

**Comment by Noisyfox** [comment]
> > BTW that seems to be a separate problem that not really related to this feature. Will look into that soon.
> 
> yes is no issue from this feature. the problem is that generally 100% overhangs use the bridge speed, it is also in the tooltip. speed
> 
> Here a 100% overhang it use the Bridge speed bri

https://github.com/SoftFever/OrcaSlicer/issues/1749
Just to confirm: Is this the same issue as yours?

**Comment by tome9111991** [comment]
> 
> #1749 Just to confirm: Is this the same issue as yours?

not 100% sure, but could be the same or very similar problem.

**Pull request merged**